src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -58,76 +58,71 @@
    *
    * @return the info for the health endpoint regarding the distribution just performed
    */
-  public final DistributionInfo distribute() {
+  final DistributionInfo distribute() {
     log.debug("ActionDistributor awoken...");
     DistributionInfo distInfo = new DistributionInfo();
 
-    try {
-      List<ActionType> actionTypes = actionTypeRepo.findAll();
-
-      if (!CollectionUtils.isEmpty(actionTypes)) {
-        for (ActionType actionType : actionTypes) {
-          log.debug("Dealing with actionType {}", actionType.getName());
-          int successesForActionRequests = 0;
-          int successesForActionCancels = 0;
-
-          List<Action> actions = null;
-          try {
-            actions = retrieveActions(actionType);
-          } catch (Exception e) {
-            log.error("Failed to obtain actions - error msg {} - cause {}", e.getMessage(), e.getCause());
-            log.error("Stacktrace: ", e);
-          }
-
-          if (!CollectionUtils.isEmpty(actions)) {
-            log.debug("Dealing with actions {}", actions.stream().map(a -> a.getActionPK().toString()).collect(
-                Collectors.joining(",")));
-            for (Action action : actions) {
-              try {
-                if (action.getState().equals(ActionDTO.ActionState.SUBMITTED)) {
-                  actionProcessingService.processActionRequest(action);
-                  successesForActionRequests++;
-                } else if (action.getState().equals(ActionDTO.ActionState.CANCEL_SUBMITTED)) {
-                  actionProcessingService.processActionCancel(action);
-                  successesForActionCancels++;
-                }
-              } catch (Exception e) {
-                log.error("Exception {} thrown processing action {}. Processing will be retried at next scheduled "
-                        + "distribution", e.getMessage(), action.getId());
-                log.error("Stacktrace: ", e);
-              }
-            }
-
-            try {
-              actionDistributionListManager.deleteList(actionType.getName(), true);
-            } catch (LockingException e) {
-              log.error("Failed to remove the list of actions just processed from distributed list - "
-                  + "actions distributed OK, but underlying problem may remain");
-              log.error("Stacktrace: ", e);
-            }
-          }
-
-          try {
-            actionDistributionListManager.unlockContainer();
```

Comment:
Do we need to still try to unlock the container or will it unlock on it's own? is a try with resources appropriate here?

Reply:
`actionDistributionListManager.deleteList(actionType.getName(), true);` will also unlock the container. 

try with resources would be nice but DistributedListManager doesn't implement Closeable but not to worry I am removing DistributedListManager from this logic